src/argilla/client/feedback/training/schemas.py

```diff
@@ -1064,13 +1082,136 @@ def _dict_to_format(ds: datasets.Dataset) -> List[Dict[str, List[Dict[str, str]]
             return _dict_to_format(ds)
 
 
+class TrainingTaskForSentenceSimilarityFormat(BaseModel):
+    r"""
+    Union[
+        Dict[str, Union[float, int]],  # case 1 with with two string elements and one int/float, case 3 with one or three strings and one int/float.
+        Dict[str, str],                # case 2 with two elements, case 4 with three elements
+    ]
+
+    For a reference of the different cases take a look at:
+    https://huggingface.co/blog/how-to-train-sentence-transformers#how-to-prepare-your-dataset-for-training-a-sentence-transformers-model
+    """
+
+    format: Union[Dict[str, Union[float, int]], Dict[str, str]]
+
+
+class TrainingTaskForSentenceSimilarity(BaseModel, TrainingData):
+    """Training data for sentence similarity.
+
+    Args:
+        formatting_func: A formatting function converting a dictionary of records into
+            a dictionary of a pair of sentences, a pair of sentences and a label,
+            a sentence and a label or a triplet of sentences.
+
+    Examples:
+        Example for argilla/emotion dataset:
+        >>> from argilla import TrainingTaskForSentenceSimilarity
+        >>> dataset = rg.FeedbackDataset.from_argilla(name="argilla/emotion")
+        >>> def formatting_func(sample: Dict[str, Any]):
+        ...     return {"sentence": sample["text"], "label": int(sample["label"][0]["value"])}
+        >>> task = TrainingTaskForSentenceSimilarity(formatting_func=formatting_func)
+        >>> dataset.prepare_for_training(framework="...", task=task)
+    """
+
+    _formatting_func_return_types = TrainingTaskForSentenceSimilarityFormat
+    formatting_func: Callable[[Dict[str, Any]], Union[None, Tuple[str, str, str], Iterator[Tuple[str, str, str]]]]
+
+    def _format_data(self, dataset: "FeedbackDataset") -> List[Dict[str, Any]]:
+        outputs = []
+        for sample in dataset:
+            output = self.formatting_func(sample)
+            if output is None:
+                continue
+
+            self._test_output_formatting_func(output)
+
+            outputs.append(output)
+
+        return outputs
+
+    @property
+    def supported_frameworks(self):
+        names = ["sentence-transformers"]
+        return [Framework(name) for name in names]
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}\n\t formatting_func={self.formatting_func}"
+
+    @requires_version("scikit-learn")
+    def _train_test_split(self, data: List[dict], train_size: float, seed: int) -> Tuple[List[dict], List[dict]]:
+        from sklearn.model_selection import train_test_split
+
+        return train_test_split(
```

Comment:
Should we consider stratifying this split too? I can imagine the model might over-under estimate when not doing this?

Reply:
Sure, I will add it for the datasets that include integer labels.